src/components/ReactTags.tsx

```diff
@@ -132,6 +151,16 @@ const ReactTags = (props: ReactTagsProps) => {
     if (props.handleFilterSuggestions) {
       return props.handleFilterSuggestions(query, updatedSuggestions);
     }
+    if (enableFuzzySearch) {
+      const newSuggestionsFuzzy = fuzzySuggestions.search(query);
+
+      return props.suggestions.filter((suggestion) =>
+        newSuggestionsFuzzy.find(
+          (sug) =>
+            sug.text === suggestion.id && sug.distance < maximumFuzzyDistance
+        )
+      );
```

Comment:
This should be using **`updatedSuggestions`** instead of props.suggestions to retain the modifications made by the code above this
This can lead to bugs like showing the suggestions which are already used when `allowUnique` is true and fuzzy search is enabled.

Here is a quick repro - note "India" is shown in suggestion, even though its used already and we pass "allowUnique" as true in example and fuzzy search is enabled.

![image](https://github.com/user-attachments/assets/bfac8254-20ab-479b-90b8-fdf7324ee4fe)

Additionally this code should be **before**  👇 code
```
 // If a custom filter function is provided, use it to filter the suggestions
    if (props.handleFilterSuggestions) {
      return props.handleFilterSuggestions(query, updatedSuggestions);
    }
    ```
    
    So filteredSuggestions prop has correct suggestions when host wants to handle it.

Reply:
make sense, I will update it 👍

Reply:
[refactor: fuzzySuggestions updation logic](https://github.com/react-tags/react-tags/pull/985/commits/c4a741bf8512a52c265d86c24cb3282fe22fd008) done 

using updated suggestions over props.suggestions 
https://github.com/react-tags/react-tags/blob/c4a741bf8512a52c265d86c24cb3282fe22fd008/src/components/ReactTags.tsx#L136

move fuzzy search logic above custom logic by the user 
https://github.com/react-tags/react-tags/blob/c4a741bf8512a52c265d86c24cb3282fe22fd008/src/components/ReactTags.tsx#L154

![image](https://github.com/user-attachments/assets/e9d92f57-972d-42d5-8004-63b5ba5f5e7a)